crates/miden-lib/asm/miden/contracts/auth/multisig.masm

```diff
@@ -0,0 +1,292 @@
+use.miden::account
+use.miden::auth
+use.miden::tx
+use.std::crypto::dsa::rpo_falcon512
+use.std::crypto::hashes::rpo
+use.std::sys
+
+# CONSTANTS
+# =================================================================================================
+
+# Auth Request Constants
+
+# The event to request an authentication signature.
+const.AUTH_REQUEST=131087
+
+# The event emitted when a signature is not found for a required signer.
+const.UNAUTHORIZED_EVENT=131101
+
+# Storage Layout Constants
+#
+# ┌───────────┬───────────────┬──────────┬──────────────┐
+# │ THRESHOLD │ NUM APPROVERS │ PUB KEYS │ EXECUTED TXS │
+# │  (value)  │    (value)    │   (map)  │    (map)     │
+# ├───────────┼───────────────┼──────────┼──────────────┤
+# │     0     │       1       │    2     │      3       │
+# └───────────┴───────────────┴──────────┴──────────────┘
+
+# The slot in this component's storage layout where the signature threshold is stored.
+const.THRESHOLD_SLOT=0
+
+# The slot in this component's storage layout where the number of approvers is stored.
+const.NUMBER_OF_APPROVERS_SLOT=1
+
+# The slot in this component's storage layout where the public keys map is stored.
+const.PUBLIC_KEYS_MAP_SLOT=2
+
+# The slot in this component's storage layout where executed transactions are stored.
+const.EXECUTED_TXS_SLOT=3
+
+# Runtime Memory Address Constants
+const.TEMP_PK=108
+const.MESSAGE=112
+const.SUCCESSFUL_VERIFICATIONS=116
+const.THRESHOLD=120
+const.NUM_OF_APPROVERS=124
+
+# Executed Transaction Flag Constant
+const.USED_FLAG=1
+
+# ERRORS
+const.TX_ALREADY_EXECUTED="tx hash exists in tx map"
```

Comment:
I would make this error more detailed. If someone receives just that error from an account with multiple account components, it would not be obvious which component failed.

Maybe something like "failed to approve multisig transaction as it was already executed".